Commands/PublishCommand.cs

```diff
@@ -65,52 +65,260 @@ public static int PublishFile(PublishOptions options, Config.Config config, stri
                 Console.WriteLine(Red("Exiting"));
                 return 1;
             }
+            
             using var client = new HttpClient();
+            client.DefaultRequestHeaders.Authorization = config.GetAuthHeader();
 
+            var uploadRequest = new HttpRequestMessage(HttpMethod.Post, $"{config.PublishConfig.Repository}/api/experimental/usermedia/initiate-upload/")

```

Comment:
This function is massive after the changes here, could it maybe be split into smaller parts that make sense? The file upload at least is a clearly separate step from the submission.